include/jwt-cpp/jwt.h

```diff
@@ -469,24 +469,54 @@ namespace jwt {
 			}
 #endif
 		};
+
+		struct unique_bio_ptr : std::unique_ptr<BIO, decltype(&BIO_free_all)> {
+			unique_bio_ptr() : std::unique_ptr<BIO, decltype(&BIO_free_all)>(BIO_new(BIO_s_mem()), BIO_free_all) {}
+			unique_bio_ptr(const std::string& data)
+				: std::unique_ptr<BIO, decltype(&BIO_free_all)>(
+#if OPENSSL_VERSION_NUMBER <= 0x10100003L
+					  BIO_new_mem_buf(const_cast<char*>(data.data()), static_cast<int>(data.size())), BIO_free_all
+#else
+					  BIO_new_mem_buf(data.data(), static_cast<int>(data.size())), BIO_free_all
+#endif
+				  ) {
+			}
+
+			unique_bio_ptr(const unique_bio_ptr&) = delete;
+			unique_bio_ptr(unique_bio_ptr&&) noexcept = default;
+			~unique_bio_ptr() noexcept = default;
+		};
+
+		struct unique_evp_md_ctx_ptr :
+#ifdef JWT_OPENSSL_1_0_0
+			std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_destroy)>
+#else
+			std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)>
+#endif
+		{
+			unique_evp_md_ctx_ptr()
+				:
+#ifdef JWT_OPENSSL_1_0_0
+				  std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_destroy)>(EVP_MD_CTX_create(), &EVP_MD_CTX_destroy)
+#else
+				  std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)>(EVP_MD_CTX_new(), EVP_MD_CTX_free)
+#endif
```

Comment:
See above. `EVP_MD_CTX_destroy` and `EVP_MD_CTX_free` have the same signature, so we can just use `void(*)(EVP_MD_CTX*)` as the destructor type (which is what both decltypes would resolve to anyway. Also in the `EVP_MD_CTX_free` branch theres a missing `&` at the end.